### PR TITLE
fix: Fix getProviderState networkVersion for request queue

### PR DESCRIPTION
**Pull request by jiexi**
## **Description**

Currently the globally selected network's networkId is being sent as the `networkVersion` to all dapps when request queue is enabled. This PR fixes this by making an appropriate `net_version` call when each dapp's inpage provider requires it.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23590?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2176

## **Manual testing steps**

1. Enable request queue
2. Go to a dapp (Refresh the page if you had it opened before request queue was on)
3. use `wallet_switchEthereumChain` to make the dapp and globally selected networks different
3. check `window.ethereum.chainId` and `window.ethereum.networkVersion` values, they should match the dapp's network
1. Disable request queue
2. Go to a dapp (Refresh the page if you had it opened before request queue was on)
3. check `window.ethereum.chainId` and `window.ethereum.networkVersion` values, they should match the global network

### **Before**

Request Queue On

https://github.com/MetaMask/metamask-extension/assets/918701/6edde8d0-c911-4349-91f2-e7b5a18b32a8

### **After**

Request Queue On

https://github.com/MetaMask/metamask-extension/assets/918701/3490e442-4562-458d-b138-59cb7c6f34b9

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -5299,8 +5269,12 @@ export default class MetamaskController extends EventEmitter {
...
     Object.keys(this.connections).forEach((origin) => {
       Object.values(this.connections[origin]).forEach(async (conn) => {
-        if (conn.engine) {
-          conn.engine.emit('notification', await getPayload(origin));
+        try {
+          if (conn.engine) {
+            conn.engine.emit('notification', await getPayload(origin));
+          }
+        } catch (err) {
+          console.error(err);
```
Comment (adonesky1) [inline app/scripts/metamask-controller.js:5277]: why is this try/catch necessary now?

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -2676,20 +2669,21 @@ export default class MetamaskController extends EventEmitter {
     // subset of state for metamask inpage provider
     const publicConfigStore = new ObservableStore();
 
-    const selectPublicState = (chainId, { isUnlocked }) => {
+    const selectPublicState = async ({ isUnlocked }) => {
```
Comment (adonesky1) [inline app/scripts/metamask-controller.js:2672]: The description of this method says:
```
Constructor helper: initialize a public config store.
This store is used to make some config info available to Dapps synchronously.
```
Can we still say this is made available synchronously?

**Comment by adonesky1** [comment]
It seems like we may now be breaking the UX assumption of [this test](https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/75206/workflows/f8ad0af7-7da5-482d-bfea-d7ce13488ccd/jobs/2578762/tests): 
`doesn't make any network requests to infura before onboarding is completed`

File: app/scripts/metamask-controller.js (hunk 2 above)
Comment (jiexi) [reply app/scripts/metamask-controller.js:2672]: > This store is used to make some config info available to Dapps synchronously.

I believe this is referring to the state that the inpage provider exposes to the dapp in a synchronous manner

File: app/scripts/metamask-controller.js (hunk 1 above)
Comment (jiexi) [reply app/scripts/metamask-controller.js:5277]: in the event that the payload function throws an error

**Comment by jiexi** [comment]
`doesn't make any network requests to infura before onboarding is completed` scenario should be resolved here [c283e37](https://github.com/MetaMask/metamask-extension/pull/23590/commits/c283e374bbf89e1d8d3bb6d4899453fa540add68)

File: app/scripts/metamask-controller.js (hunk 1 above)
Comment (BelfordZ) [reply app/scripts/metamask-controller.js:5277]: oh yeah if the await rejects its unhandled promise rejection

**Review by BelfordZ — APPROVED** [review]

**Review by adonesky1 — APPROVED** [review]
LGTM

**Pull request merged**